engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -355,6 +362,17 @@ def scc_activate_product
           end
           logger.info 'No token provided' if params[:token].blank?
         end
+
+        def de_register_hybrid(auth)
+          response = SccProxy.deregister_system_scc(auth, @system.system_token)
+          unless response.code_type == Net::HTTPNoContent
+            error = JSON.parse(response.body)
+            logger.info "Could not de-activate system #{@system.login}, error: #{error['error']} #{response.code}"
+            error['error'] = SccProxy.parse_error(error['error'], params[:token], params[:email]) if error['error'].include? 'json'
+            raise ActionController::TranslatedError.new(error['error'])
+          end
+          logger.info 'System successfully deregistered from SCC'
```

Comment:
How can the system be treated as "successfully deregistered" regardless of the response from SCC?

Shouldn't this be in a condition based on some positive response from SCC?

Reply:
according to the [API documentation](https://scc.suse.com/connect/v4/documentation#/systems/delete_systems), that's the response 
- code 204
- No content

so checking [Net::HTTPNoContent](https://docs.ruby-lang.org/en/master/Net/HTTPNoContent.html) seems appropiate to me